src/python/pants/backend/terraform/dependency_inference.py

```diff
@@ -139,9 +143,46 @@ async def infer_terraform_module_dependencies(
     return InferredDependencies(terraform_module_addresses)
 
 
+@dataclass(frozen=True)
+class TerraformDeploymentDependenciesInferenceFieldSet(TerraformDeploymentFieldSet):
+    pass
+
+
+class InferTerraformDeploymentDependenciesRequest(InferDependenciesRequest):
+    infer_from = TerraformDeploymentDependenciesInferenceFieldSet
+
+
+@rule
+async def infer_terraform_deployment_dependencies(
+    request: InferTerraformDeploymentDependenciesRequest,
+) -> InferredDependencies:
+    root_module_address_input = request.field_set.root_module.to_address_input()
+    root_module = await Get(Address, AddressInput, root_module_address_input)
+
+    # TODO: add depenendency on the vars files
+    #  The use of this target_name here leads to a dependency on self.
+    #  Is there a way to avoid the reference? Would generating targets for the vars files help?
```

Comment:
I'm still confused. Do we not generate per-file targets for the var_files sources, as we would for a field literally named "sources"? If so, these are distinct targets from self, no?

Reply:
lemme work up the Backend and Vars as separate targets